py3status/modules/wwan_status_nm.py

```diff
@@ -0,0 +1,135 @@
+        is a 4G connection.
+        (default False)
+    format_down: What to display when the modem is not plugged in.
+        (default 'WWAN: {operator} {netgen} ({signal})')
+    format_error: What to display when modem can't be accessed.
+        (default 'WWAN: {error}')
+    format_up: What to display upon regular connection
+        (default 'WWAN: {operator} {netgen} ({signal})')
+    modem: The modem device to use.
+        (default 'auto')
+
+Color options:
+    color_bad: Error or no connection
+    color_degraded: Low generation connection eg 2G
+    color_good: Good connection
+
+Requires:
+    ModemManager
+    NetworkManager
+    pydbus
+
+@author Cyril Levis <levis.cyril@gmail.com>
+
+SAMPLE OUTPUT
+{'color': '#00FF00', 'full_text': u'Bouygues Telecom 4G (19%)'}
+
+off
+{'color': '#FF0000', 'full_text': u'Bouygues Telecom 4G (12%)'}
+"""
+
+from pydbus import SystemBus
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    cache_timeout = 5
+    consider_3G_degraded = False
+    format_down = 'WWAN: {operator} {netgen} ({signal})'
+    format_error = 'WWAN: {error}'
+    format_up = 'WWAN: {operator} {netgen} ({signal})'
+    """
+    modem: 'auto' value will use first find modem or
+    use 'busctl introspect org.freedesktop.ModemManager1 \
+            /org/freedesktop/ModemManager1/Modem/0'
+    and read '.EquipmentIdentifier')
+    """
+    modem = 'auto'
+
+    def wwan_status_nm(self):
+        response = {}
+        response['cached_until'] = self.py3.time_in(self.cache_timeout)
+
+        bus = SystemBus()
+        for id in range(0, 20):
+            """
+            find the modem
+            """
+            device = 'Modem/' + str(id)
+
+            try:
+                proxy = bus.get('.ModemManager1', device)
+
+                eqid = str(proxy.EquipmentIdentifier)
+
+                if (self.modem != 'auto'
+                        and eqid == self.modem) or (self.modem == 'auto'):
+                    state = proxy.State
+                    signal = str(proxy.SignalQuality[0]) + '%'
+                    modes = proxy.CurrentModes[0]
+                    operator = proxy.OperatorName
+                    netgen = self._get_capabilities(modes)
+
+                    if state == 11:
+                        response['full_text'] = self.py3.safe_format(
+                            self.format_up,
+                            dict(
+                                netgen=netgen[1],
+                                signal=signal,
+                                operator=operator))
+                        """
+                        green color if 4G, else yellow
+                        """
+                        if netgen[0] == 4:
+                            response['color'] = self.py3.COLOR_GOOD
+                        elif netgen[0] == 3 and self.consider_3G_degraded is False:
+                            response['color'] = self.py3.COLOR_GOOD
```

Comment:
By using `thresholds` (see #1285 for an example), users could just do `thresholds = [(3, 'good'), (4, 'degraded')]`. Users also can adjust thresholds instead of toggling `consider_3G_degraded too`. More flexible (and clean) this way.